redact_pii.c

```diff
@@ -0,0 +1,13 @@
+#include <string.h> 
+int redact_sip_pii = 0;
+
+char* redact_pii(char* input) {
```

Comment:
Is there a reason why you can't use the zsw and zsrw macros? In this case you will cause side effects by overwriting the input string which WILL cause side effects, example being here https://github.com/purecloudlabs/opensips/pull/51/files#diff-26259bd8f72b9eabb42132fea3b2bc2dc04923b6ba94ea99c479e8e45fe9fed7R4318 the uri string is now ****

In C if you have a char* as an input it's a pointer to a block of memory and when you do something like strcpy like above you're overwriting the memory, so if the input in this case was `my-sip-uri.com` it is not `****ip-uri.com` also if the uri is `a.ie` then you will cause undefined behaviour which will likely result in a crash or other undesired behaviour because you're overwriting memory that is belonging to this pointer because the string **** is four bytes and the input is only three.

I can't see the original definition of the zsrw macro but is there no reason why we can't do something like this

`char* redact_pii(const char* input) {
    if (input == NULL) {
        return "";
    }

    if (redact_sip_pii) {
        return ZSRW(input);
    }

    return ZSW(input);
}`

Reply:
Also, if we use macros that already check for `NULL`, we can simplify the code (and avoid unnecessary `NULL` checks):
```c
inline char* redact_pii(const char* input) {
    return redact_sip_pii ? ZSRW(input) : ZSW(input);
}
```
You can give an inline hint to the compiler, though it may not necessarily apply it. In terms of performance, duplicating the function body on the caller's side might be more efficient than creating a function call if the function body is short enough.

Reply:
Probably don't need the ZSRW macro because all it does is check the input if it's null, even simpler

`inline const char* redact_pii(const char* input) {
    return redact_sip_pii ? "***" : ZSW(input);
}`

Reply:
It slightly changes the output (compared to the first implementation with macros) because if the input is NULL, the original behavior is to log an empty string, which may have a value indicating that something is empty. However, this depends on whether we need this behavior or not. I'm OK with both.

Reply:
I think keeping the empty string could be important. Logging `***` for an empty string is very misleading, and given how far down this is in the logging code, it would be very difficult for someone to realize that there wasn't actually anything to log.